File: back/src/main/java/org/example/skillandyou/entity/Review.java
package org.example.skillandyou.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "reviews")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "exchange_id", nullable = false)
    private Long exchangeId;

    @Column(name = "reviewer_id", nullable = false)
    private Long reviewerId;

    @Column(name = "reviewed_user_id", nullable = false)
    private Long reviewedUserId;

    @Column(nullable = false)
    private Integer rating;

    @Column(columnDefinition = "TEXT")
    private String comment;

    private LocalDateTime createdAt = LocalDateTime.now();
}

File: front/src/types/Review.ts
export interface Review {
    id: number;
    exchangeId: number;
    reviewerId: number;
    reviewedUserId: number;
    rating: number;
    comment: string;
    createdAt: string;
}

export interface ReviewRequest {
    rating: number;
    comment: string;
}

export interface UserRating {
    userId: number;
    averageRating: number;
    totalReviews: number;
}
